app/models/miq_server/worker_management/heartbeat.rb

```diff
@@ -91,11 +91,23 @@ def validate_heartbeat(w)
     end
   end
 
+  def clean_heartbeat_files
+    Dir.glob(Rails.root.join("tmp", "*.hb")).each { |f| File.delete(f) }
+  end
+
   private
 
-  def workers_last_heartbeat(pid)
+  def workers_last_heartbeat(w)
+    ENV["WORKER_HEARTBEAT_METHOD"] == "file" ? workers_last_heartbeat_to_file(w) : workers_last_heartbeat_to_drb(w)
+  end
+
+  def workers_last_heartbeat_to_drb(w)
     @workers_lock.synchronize(:SH) do
-      @workers.fetch_path(pid, :last_heartbeat)
+      @workers.fetch_path(w.pid, :last_heartbeat)
     end
   end
+
+  def workers_last_heartbeat_to_file(w)
+    File.mtime(w.heartbeat_file).utc
```

Comment:
Good thought @jrafanie. I'll add that check because I don't think it should just assume it exists.